### ISSUE TITLE
Bit-arrays bug

### ISSUE THREAD

**Issue by nomennescio** [factor/factor#2909]
```factor
M: bit-array equal?
    over bit-array? [ [ underlying>> ] bi@ sequence= ] [ 2drop f ] if ;
```

by not checking `length` this might accidentally compare equal

**Comment by mrjbq7** [comment]
``sequence=`` checks length, but perhaps you mean this should return ``t``:

```factor
IN: scratchpad 2 B{ 0b11 } bit-array boa 
2 B{ 0b111 } bit-array boa = .
f
```

**Comment by nomennescio** [comment]
`sequence=` checks the length of the *underlying* byte array, but what's missing is the check on the bit array length:```
3 B{ 0b0 } bit-array boa
2 B{ 0b0 } bit-array boa = .
t
```

**Comment by mrjbq7** [comment]
Okay two bugs!On Oct 31, 2023, at 12:59 PM, nomennescio ***@***.***> wrote:﻿
sequence= checks the length of the underlying byte array, but what's missing is the check on the bit array length:```
3 B{ 0b0 } bit-array boa
2 B{ 0b0 } bit-array boa = .
t

**Comment by mrjbq7** [comment]
Fixed in fd09fc222233e06d1f8c7a915a7a1d43e16456eb.
